test/python/result/test_mitigators.py

```diff
@@ -491,5 +425,104 @@ def test_error_handling(self):
+        ]
+        expected_mitigation_matrix = np.array(
+            [
+                [
+                    1.03929190e00,
+                    -4.33478790e-02,
+                    -3.13669642e-02,
+                    1.30828632e-03,
+                    -2.60083030e-02,
+                    1.08478164e-03,
+                    7.84958982e-04,
+                    -3.27398944e-05,
+                ],
+                [
+                    -1.26871849e-02,
+                    1.06995259e00,
+                    3.82913092e-04,
+                    -3.22923375e-02,
+                    3.17497086e-04,
+                    -2.67755876e-02,
+                    -9.58240871e-06,
+                    8.08116468e-04,
+                ],
+                [
+                    -1.04556476e-02,
+                    4.36095141e-04,
+                    1.06020322e00,
+                    -4.42200702e-02,
+                    2.61652816e-04,
+                    -1.09132907e-05,
+                    -2.65316092e-02,
+                    1.10660825e-03,
+                ],
+                [
+                    1.27637610e-04,
+                    -1.07641051e-02,
+                    -1.29424604e-02,
+                    1.09148083e00,
+                    -3.19413406e-06,
+                    2.69371972e-04,
+                    3.23885361e-04,
+                    -2.73143321e-02,
+                ],
+                [
+                    -1.66453157e-02,
+                    6.94260323e-04,
+                    5.02373800e-04,
+                    -2.09535346e-05,
+                    1.04865489e00,
+                    -4.37384003e-02,
+                    -3.16495494e-02,
+                    1.32007268e-03,
+                ],
+                [
+                    2.03198156e-04,
+                    -1.71363778e-02,
+                    -6.13274220e-06,
+                    5.17194593e-04,
+                    -1.28014838e-02,
+                    1.07959180e00,
+                    3.86362759e-04,
+                    -3.25832593e-02,
+                ],
+                [
+                    1.67457819e-04,
+                    -6.98450675e-06,
+                    -1.69802316e-02,
+                    7.08229351e-04,
+                    -1.05498426e-02,
+                    4.40023925e-04,
+                    1.06975459e00,
+                    -4.46184491e-02,
+                ],
+                [
+                    -2.04424601e-06,
+                    1.72398080e-04,
+                    2.07286652e-04,
+                    -1.74811743e-02,
+                    1.28787498e-04,
+                    -1.08610790e-02,
+                    -1.30590591e-02,
+                    1.10131398e00,
+                ],
+            ]
+        )
+        LRM = LocalReadoutMitigator(assignment_matrices, qubits)
+        self.assertTrue(matrix_equal(expected_mitigation_matrix, LRM.mitigation_matrix()))
```

Comment:
This is pretty opaque - how do you know that this is exactly correct?  It'd be best if we could have a simple analytic case to check, but failing that, a comment about how we know this is correct would be good.

Reply:
I know since I generated those numbers... Not using a concrete number-based test case makes us vulnerable to errors in the testing code itself, and makes it easier to accidently "fix" the test code to pass when there's a bug in the tested code. Most of the test code is based on computations we do on the fly but I believe some hard-coded numbers are a safeguard we need.